src/ui/common/src/components/pages/LoginPage.tsx

```diff
@@ -36,18 +41,25 @@ export const LoginPage: React.FC = () => {
     if (!apiKey || apiKey.length === 0) {
       setValidationError(true);
       setErrorMsg('Api key should not be empty.');
+      setIsAuthed(false);
     } else if (!success) {
       setValidationError(true);
       setErrorMsg(
         "Invalid key, please copy the key from 'aqueduct apikey' outputs."
       );
+      setIsAuthed(false);
     } else {
       setValidationError(false);
       setUser(apiKey);
-      navigate('/');
+      setIsAuthed(true);
+      navigate('/', { replace: true })
```

Comment:
This and below are pretty much the same, but was struggling with getting things to work so tried hooks and component way of doing this.